lib/decimal.ex

```diff
@@ -1402,10 +1402,6 @@ defmodule Decimal do
     end
   end
 
-  defp do_reduce(0, _exp) do
-    %Decimal{coef: 0, exp: 0}
-  end
-
   defp do_reduce(coef, exp) do
```

Comment:
if we're to be paranoid this could be: `... when coef != 0` but probably ok to leave this as is. LGTM!